### ISSUE TITLE
Probable race condition for Postgres write_to_source implementation

### ISSUE THREAD

**Issue by chamini2** [fal-ai/fal#327]
**Describe the bug**
During integration tests sometimes we get an error:

```
dbt.exceptions.InternalException: Cache inconsistency detected: in rename, new key _ReferenceKey(database='test', schema='dbt_fal', identifier='other_model') already in cache: [_ReferenceKey(database='test', schema='dbt_fal', identifier='other_model'), _ReferenceKey(database='test', schema='dbt_fal', identifier='raw_orders'), _ReferenceKey(database='test', schema='dbt_fal', identifier='zendesk_ticket_data'), _ReferenceKey(database='test', schema='dbt_fal', identifier='some_model'), _ReferenceKey(database='test', schema='dbt_fal', identifier='raw_customers'), _ReferenceKey(database='test', schema='dbt_fal', identifier='agent_wait_time'), _ReferenceKey(database='test', schema='dbt_fal', identifier='some_source'), _ReferenceKey(database='test', schema='dbt_fal', identifier='raw_payments'), _ReferenceKey(database='test', schema='dbt_fal', identifier='other_model__f__'), _ReferenceKey(database='test', schema='dbt_fal', identifier='time_series'), _ReferenceKey(database='test', schema='dbt_fal', identifier='model_e'), _ReferenceKey(database='test', schema='dbt_fal', identifier='raw_zendesk_ticket_data'), _ReferenceKey(database='test', schema='dbt_fal', identifier='model_c'), _ReferenceKey(database='test', schema='dbt_fal', identifier='model_a'), _ReferenceKey(database='test', schema='dbt_fal', identifier='customers'), _ReferenceKey(database='test', schema='dbt_fal', identifier='model_b'), _ReferenceKey(database='test', schema='dbt_fal', identifier='stg_customers'), _ReferenceKey(database='test', schema='dbt_fal', identifier='stg_payments'), _ReferenceKey(database='test', schema='dbt_fal', identifier='new_model'), _ReferenceKey(database='test', schema='dbt_fal', identifier='stg_orders')]
```

This happens in the test:

```
  Scenario: Use write_to_source function with mode append and overwrite          # features/write_to_source_function.feature:5
```

**Your environment**
- OS: ubuntu
- Paste the following commands output:
  ```
  python 3.8, dbt 0.20.1, fal@main
  ```
- Adapter being used: postgres

**How to reproduce**
It's right there: https://github.com/fal-ai/fal/runs/6578623088?check_suite_focus=true#logs

**Expected behavior**
No error

**Actual behavior**
Error

**Additional context**
I have not seen it for other adapters yet.

**Is there an existing bug report for this?**
 * [X]  I have searched the existing issues

**Comment by turbo1912** [comment]
Potential way to mitigate the issue 

https://github.com/fal-ai/fal/commit/cea45838af997aa345b9865f4b1c24d6056aeb5e

**Comment by chamini2** [comment]
It seems we really fixed this with #429
